terraform/modules/alertmanager/nlb.tf

```diff
@@ -30,18 +30,26 @@ resource "aws_acm_certificate" "alertmanager_cert" {
 }
 
 resource "aws_route53_record" "alertmanager_cert_validation" {
-  count      = 1 + length(data.aws_availability_zones.available.names)
-  name       = aws_acm_certificate.alertmanager_cert.domain_validation_options[count.index]["resource_record_name"]
-  type       = aws_acm_certificate.alertmanager_cert.domain_validation_options[count.index]["resource_record_type"]
-  zone_id    = local.zone_id
-  records    = [aws_acm_certificate.alertmanager_cert.domain_validation_options[count.index]["resource_record_value"]]
-  ttl        = 60
+  for_each = {
```

Comment:
I'm not entirely convinced this will match the `1 +` logic so I'd appreciate someone familiar with this to check this carefully.

Reply:
I think this is fine as long as it works at all.  The `1 + length` logic is just an expression which should match the expected size of `aws_acm_certificate.alertmanager_cert.domain_validation_options` without having to dynamically depend on it, because such dynamic dependencies made terraform sad.

Reply:
see https://www.terraform.io/docs/configuration/expressions.html#values-not-yet-known

> The count meta-argument for resources cannot be unknown, since it must be evaluated during the plan phase to determine how many instances are to be created.

Reply:
:+1: